### PR TITLE
Implement a contact.us handler to record emails

### PR DESCRIPTION
**Pull request by jwhite66**
This will enable the front end /contact form to operate.

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -48,6 +48,29 @@ def send_thank_you(name, email, url_nonce, amount_cents):
   message.html = open('email/thank-you.html').read().format(**format_kwargs)
   message.send()
 
+class ContactUsHandler(webapp2.RequestHandler):
+  def post(self):
+    data = json.loads(self.request.body)
+    sender = data["name"] + " <" + data["email"] + ">";
```
Comment (hjfreyer) [inline backend/main.py:54]: Use the % operator

File: backend/main.py [hunk 2]
```diff
@@ -48,6 +48,29 @@ def send_thank_you(name, email, url_nonce, amount_cents):
...
+    message = mail.EmailMessage(sender=sender, subject=data['subject'])
+    message.to = "info@mayone.us"
+    message.body = data['body']
+    message.send()
+    logging.info("Contact from: " + sender + "; regarding " + data['subject'])
+    _origin = self.request.headers['Origin']
+    self.response.headers.add_header("Access-Control-Allow-Origin", _origin)
```
Comment (hjfreyer) [inline backend/main.py:61]: Can you pull this into an enable_cors() function?

File: backend/main.py [hunk 3]
```diff
@@ -48,6 +48,29 @@ def send_thank_you(name, email, url_nonce, amount_cents):
...
+    self.response.headers.add_header("Access-Control-Allow-Methods", "POST, OPTIONS")
+    self.response.headers.add_header("Access-Control-Allow-Credentials", "true")
+    self.response.headers.add_header("Access-Control-Allow-Headers", "authorization, origin, x-requested-with, content-type, accept")
+    self.response.write('Ok.')
+
+  def options(self):
```
Comment (hjfreyer) [inline backend/main.py:67]: Do you know if our use case here actually uses the preflight? I thought it was only for sufficiently weird requests.

File: backend/main.py [hunk 4]
```diff
@@ -182,7 +205,8 @@ def post(self, url_nonce):
   ('/stripe_public_key', GetStripePublicKeyHandler),
   ('/pledge.do', PledgeHandler),
   ('/user-update/(\w+)', UserUpdateHandler),
-  ('/campaigns/may-one/?', EmbedHandler)
+  ('/campaigns/may-one/?', EmbedHandler),
+  ('/contact.us', ContactUsHandler),
```
Comment (hjfreyer) [inline backend/main.py:209]: I kinda like the "contact.do" convention for things which are POST-only. "contact.us" could easily just be an HTML page.

**Comment by hjfreyer** [comment]
some notes

**Comment by jwhite66** [comment]
Good points.  I'll respin a bit later today and hope that no one else notices I'm a C programmer pretending to know Python... <grin>.

**Comment by jwhite66** [comment]
I'll close this and send a new PR with an updated version.

**Pull request closed without merging**